### PR TITLE
Test crate

### PR DESCRIPTION
**Pull request by prsabahrami**
Added a crate for testing. Created the repo on top of #124, so after merging this we can remove the other PR

### CODE REVIEW

**Comment by certik** [comment]
Thanks for adding this --- I think this now tests the actual shell execution in [crates/tests/src/lib.rs](https://github.com/prefix-dev/shell/pull/135/files#diff-08a3de0025ac1db00ab1c27ab6e2816a262c06ae1a36cc0d672778c5ab103795), correct?

**Comment by prsabahrami** [comment]
Yes! This tests shell's execution.

File: crates/tests/src/test_builder.rs [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+    expected_exit_code: i32,
+    expected_stderr: String,
+    expected_stdout: String,
+    assertions: Vec<TestAssertion>,
+    assert_stdout: bool,
+    assert_stderr: bool,
+}
+
+impl TestBuilder {
+    #[allow(clippy::new_without_default)]
```
Comment (Hofer-Julian) [inline crates/tests/src/test_builder.rs:74]: Why not just put that in default and call that then within new? :)

File: crates/deno_task_shell/Cargo.toml [hunk 2]
```diff
@@ -29,9 +29,9 @@ pest_derive = "2.7.12"
 dirs = "5.0.1"
 pest_ascii_tree = { git = "https://github.com/prsabahrami/pest_ascii_tree.git", branch = "master" }
 miette = "7.2.0"
+tempfile = "3.12.0"
+pretty_assertions = "1.0.0"
```
Comment (Hofer-Julian) [inline crates/deno_task_shell/Cargo.toml:34]: Why move them out of dev-dependencies?

**Review by Hofer-Julian — COMMENTED** [review]
Looks excellent!

Had a few minor comments :)

File: crates/deno_task_shell/Cargo.toml (hunk 2 above)
Comment (wolfv) [reply crates/deno_task_shell/Cargo.toml]: fixed!

**Pull request merged**